hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -239,6 +322,43 @@ private void initIfNeeded() {
     return result;
   }
 
+  private List<Pair<String, Option<HoodieRecord<HoodieMetadataPayload>>>> readFromBaseAndMergeWithLogRecordsForKeyPrefixes(HoodieFileReader baseFileReader,
```

Comment:
@alexeykudinkin : I tried to unify this and the other method. but there are quite a few places where I had to do if else branching. so, have left it as is. If you dont mind, can you take a stab at unifying. I have addressed every other feedback we discussed.